test/org/sagebionetworks/bridge/dynamodb/DynamoTaskTest.java

```diff
@@ -88,9 +102,39 @@ public void hasValidStatusBasedOnTimestamps() throws Exception {
         assertEquals(TaskStatus.DELETED, task.getStatus());
         
         task = new DynamoTask();
-        task.setScheduledOn(DateTime.now().minusHours(1).getMillis());
-        task.setExpiresOn(DateTime.now().plusHours(1).getMillis());
+        task.setLocalScheduledOn(now.minusHours(1));
+        task.setLocalExpiresOn(now.plusHours(1));
         assertEquals(TaskStatus.AVAILABLE, task.getStatus());
     }
     
+    /**
+     * If a timestamp is not derived from a DateTime value passed into DynamoTask, or set after construction, 
+     * then the DateTime scheduledOn and expiresOn values are null.
+     */
+    @Test
+    public void dealsTimeZoneAppropriately() {
+        DateTime dateTime = DateTime.parse("2010-10-15T00:00:00.001+06:00");
+        DateTime dateTimeInZone = DateTime.parse("2010-10-15T00:00:00.001Z");
+        
+        // Task with datetime and zone (which is different)
+        DynamoTask task = new DynamoTask();
+        // Without a time zone, getStatus() works
+        assertEquals(TaskStatus.AVAILABLE, task.getStatus());
+        // Now set some values
+        task.setScheduledOn(dateTime);
+        task.setTimeZone(DateTimeZone.UTC);
+        
+        // Scheduled time should be in the time zone that is set
+        assertEquals(DateTimeZone.UTC, task.getScheduledOn().getZone());
+        // But the datetime does not itself change (this is one way to test this)
+        assertEquals(dateTimeInZone.toLocalDateTime(), task.getScheduledOn().toLocalDateTime());
+        
+        // setting new time zone everything shifts only in zone, not date or time
+        DateTimeZone newZone = DateTimeZone.forOffsetHours(3);
+        task.setTimeZone(newZone);
+        LocalDateTime copy = task.getScheduledOn().toLocalDateTime();
+        assertEquals(newZone, task.getScheduledOn().getZone());
+        assertEquals(dateTimeInZone.toLocalDateTime(), copy);
+    }
```

Comment:
Still not seeing unit tests for setting scheduledOn and getting localScheduledOn (or vice versa).